### PR TITLE
Feature/validation and specs

### PR DESCRIPTION
**Pull request by flemton**
## Dear reviewer, it is a pleasure to request a review of my validation and specs feature before merging into dev.

Below were the changes made
- ### Add validations to models:
- ### Add unit specs for all of your models' methods and validations.
And below were the commits made on that

- [Add validations for user model](https://github.com/flemton/Blogapp/commit/1c08ff641e104acb8d22bb636c25a0fc26431164)

- [Add validations for post model](https://github.com/flemton/Blogapp/commit/f296570ca5f4e418e2b7e96f64c16dfedcf77515)

- [Add rspecs and useful helpers](https://github.com/flemton/Blogapp/commit/ba285f9a9694ea1e865500eff88eca6a494030f0)

- [run 'rails generate rspec:install'](https://github.com/flemton/Blogapp/commit/1aea9ef5b6dc500bb4ae16bc22f35209b1620da1)

- [Update rails_helper.rb](https://github.com/flemton/Blogapp/commit/95b50fe66e65eec89e8d38fa5a9e518d1a166f20)

- [Update factory_girl to factory_bot](https://github.com/flemton/Blogapp/commit/ec94f6103fa9122caf6fde10ce76699cfd5312d0)

- [Update schema.rb](https://github.com/flemton/Blogapp/commit/f56caa53cfa31ab5ffde3fafa23222d49e5e20f6)

- [Update models](https://github.com/flemton/Blogapp/commit/facd929433dc0c0d2c80384f4380ae67c8693cd1)

- [Create user_spec.rb](https://github.com/flemton/Blogapp/commit/27659e56cac0bb9c49e64c2ad64170b3cf03e617)

- [Create post_spec.rb](https://github.com/flemton/Blogapp/commit/9e303ac3d6843f9fb076b9a5e18317a3e5a8b322)

- [Create like_spec.rb](https://github.com/flemton/Blogapp/commit/12add3355c2f99aa62d7aba930a6b9f6df006d89)

- [Create comment_spec.rb](https://github.com/flemton/Blogapp/commit/343905dd14b02768920d39fe39741544f4a1d6a3)

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject(:user) do
+    User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.',
+             posts_counter: 0)
```
Comment (topeogunleye) [inline spec/models/user_spec.rb:6]: Good job on your test coverage! However, I noticed that while we have tests for negative and zero values for `posts_counter`, we don't have an explicit test for positive values. To ensure comprehensive coverage, consider adding a test case like:

```ruby
it 'is valid with a positive posts_counter value' do
    user.posts_counter = 1
    expect(user).to be_valid
end

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @flemton, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
- Your test for the presence of a `title` and its character limit aligns with the project requirements. Well done! 👍
- `CommentsCounter` and `LikesCounter` validations:
  - You've appropriately tested against negative values which ensures they are integers and not less than zero. This meets the project specifications. Great job! 🚀
- Nice touch on the test ensuring that saving a post updates the `posts_counter` of its author. This is a valuable functional test.
- Tests for negative and zero values for `posts_counter`. Great Job. 🚀 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 1 above)
Comment (flemton) [reply spec/models/user_spec.rb:6]: Thank you very much sir for the observation

Comment (topeogunleye) [reply spec/models/user_spec.rb:6]: @flemton you're welcome

**Review by topeogunleye — APPROVED** [review]
### Hi @flemton, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 

- **Positive Progress**: Great job on addressing the feedback and adding the explicit test for positive `posts_counter` values. This is a crucial step to ensuring that our model behaves as expected across all scenarios. ✔️ 
- **Comprehensive Coverage**: With the addition of this test, your coverage for the `User` model has become more thorough, which strengthens the reliability of your codebase. ✔️ 
- **Quick Turnaround**: I appreciate your prompt response to the feedback and your proactive approach in enhancing the test suite. ✔️ 

Keep up the fantastic work! Your attention to detail and dedication to improving code quality is commendable. 🚀💪
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**